Support SOCKS over Unix domain sockets

JeremyRand opened the pull request:
Fixes https://github.com/kyuupichan/aiorpcX/issues/39 ; tested with a lightly patched spesmilo/electrumx on Whonix 15 and seems to work fine with the SOCKS Unix socket that Whonix provides.

Let me know if anything needs improvement, or if you need tests added.

JeremyRand commented:
The Coveralls fails on Azure appear to not be directly related to this PR.

kyuupichan commented on aiorpcx/util.py:
```diff
@@ -181,6 +182,30 @@ def default_port(cls, port):
+
+    def __eq__(self, other):
+        # pylint: disable=protected-access
+        return self._path == other._path
```
This needs to test isinstance before testing the member

kyuupichan commented on aiorpcx/socks.py:
```diff
@@ -272,11 +272,11 @@ def _connect_response_rest(self, addr_len):
         auth is an authentication method to use when connecting, or None.
         '''
-        if not isinstance(address, NetAddress):
+        if not isinstance(address, NetAddress) and not isinstance(address, UnixAddress):
```
Please combine these.  isinstance takes a tuple to test multiple cases.

kyuupichan reviewed (changes requested):
I've recreated the Coveralls token; hopefully that issue goes away

JeremyRand commented:
Addressed feedback, thanks for catching those.  Coveralls still unhappy as before due to a missing token.  I'm pretty sure the Py38_Win pytest failure is unrelated to this PR as well (it seems to be in a completely unrelated area of the code).

kyuupichan commented:
Hi Jeremy, the issue with the missing var seems to be fixed (I renamed it; unclear what the problem was as it used to work and I didn't change anything).  So coveralls now runs.  Can you please provide 100% test coverage of the new lines of code.

JeremyRand commented:
@kyuupichan What's the preferred way to test the Unix socket code?  Tweak `FakeServer` to handle Unix sockets?  Run an actual SOCKS proxy (e.g. Tor) on a Unix socket and connect to that?  Something else?

kyuupichan commented:
Whatever is simplest for you.  I don't think you need all the tests that already exist as the socks code is network-independent.  I'm mainly interested in code coverage.  So a single test that tests a successful connection is made would suffice - the assumption being if it works for a successful connection then all the other cases have no reason not work too.  You might need to add a dummy test for one or two of the methods of the class you add, or they might all get used anyway; I'm not sure.

JeremyRand commented:
Coveralls on Azure still seems unhappy about a missing token.  :/

kyuupichan commented:
git HEAD is not - as you can see from recent changes I've made - are you building on HEAD?

JeremyRand commented:
> git HEAD is not - as you can see from recent changes I've made - are you building on HEAD?

I think so?  This PR's parent commit is d3f5ac58f9d87ff917a04a9734d978e5821c4454.  That should have the fix, right?

kyuupichan commented:
It makes no sense to me - when you run it, azure pipelines doesn't replace $(CRT) with the variable's value; when I run it, it does.  I suggest you just add the tests that pass for you and that you think give coverage, and I just merge it.
